packages/keycloak-user-management/src/components/forms/UserForm/index.tsx

```diff
@@ -1,39 +1,24 @@
 import React, { useEffect, useState, FC } from 'react';
 import { useHistory } from 'react-router';
 import { Button, Col, Row, Form, Select, Input, Radio } from 'antd';
-import { KeycloakUser, Practitioner, UserGroup } from '../../../ducks/user';
 import lang from '../../../lang';
 import { submitForm } from './utils';
-import { Dictionary } from '@onaio/utils';
 import { sendErrorNotification } from '@opensrp/notifications';
 import '../../../index.css';
-import { ATTRIBUTES_FORM_FIELD, CONTACT_FORM_FIELD } from '../../../constants';
-
-/** props for editing a user view */
-export interface UserFormProps {
-  initialValues: FormFields;
-  keycloakBaseURL: string;
-  opensrpBaseURL: string;
-  userGroups: UserGroup[];
-  extraData: Dictionary;
-  hidden?: string[];
-}
-
-/** adhoc attributes that can be applied to a user */
-export interface UserAttributes {
-  contact?: string;
-}
-
-export interface FormFields extends KeycloakUser {
-  active?: boolean;
-  userGroup?: string[];
-  practitioner?: Practitioner;
-  attributes?: UserAttributes;
-}
+import { ATTRIBUTES_FORM_FIELD, CONTACT_FORM_FIELD, FormFieldsKey, UserFormProps } from './types';
 
 const UserForm: FC<UserFormProps> = (props: UserFormProps) => {
-  const { initialValues, keycloakBaseURL, opensrpBaseURL, extraData, userGroups, hidden } = props;
-  const isHidden = (fieldName: string) => !!hidden?.includes(fieldName);
+  const {
+    initialValues,
+    keycloakBaseURL,
+    opensrpBaseURL,
+    extraData,
+    userGroups,
+    renderFields,
+    hiddenFields,
+  } = props;
+  const shouldRender = (fieldName: FormFieldsKey) => !!renderFields?.includes(fieldName);
+  const isHidden = (fieldName: FormFieldsKey) => !!hiddenFields?.includes(fieldName);
```

Comment:
I don't think we need two checks/env variables for this. isHidden === !shouldRender. REACT_APP_USER_FORM_HIDDEN_FIELDS alone should suffice

Reply:
its related but will imply different things; field can be rendered but hidden. goal is to unify this interface for all fields and forms, because some do require this functionality, i.e. fields that should be rendered but remain hidden.